contracts/lease/src/position/spec.rs

```diff
@@ -0,0 +1,117 @@
+use currency::Currency;
+use finance::{
+    coin::Coin,
+    liability::Liability,
+    price::{self, Price},
+};
+
+use crate::{
+    api::PositionSpecDTO,
+    error::{ContractError, ContractResult},
+};
+
+#[cfg_attr(test, derive(Debug))]
+pub struct Spec<Lpn> {
+    liability: Liability,
+    min_asset: Coin<Lpn>,
+    min_trasaction_amount: Coin<Lpn>,
+}
+
+impl<Lpn> Spec<Lpn>
+where
+    Lpn: Currency,
+{
+    fn new_internal(
+        liability: Liability,
+        min_asset: Coin<Lpn>,
+        min_trasaction_amount: Coin<Lpn>,
+    ) -> Self {
+        let obj = Self {
+            liability,
+            min_asset,
+            min_trasaction_amount,
+        };
+        debug_assert_eq!(Ok(()), obj.invariant_held());
+        obj
+    }
+
+    pub fn new(
+        liability: Liability,
+        min_asset: Coin<Lpn>,
+        min_trasaction_amount: Coin<Lpn>,
+    ) -> Self {
+        Self::new_internal(liability, min_asset, min_trasaction_amount)
+    }
+
+    pub fn liability(&self) -> Liability {
+        self.liability
+    }
+
+    pub fn check_trasaction_amount<Trasactional>(
+        &self,
+        amount: Coin<Trasactional>,
+        trasactional_in_lpn: Price<Trasactional, Lpn>,
+    ) -> ContractResult<()>
+    where
+        Trasactional: Currency,
+    {
+        let amount = price::total(amount, trasactional_in_lpn);
+
+        if amount < self.min_trasaction_amount {
+            Err(ContractError::PositionCloseAmountTooSmall(
```

Comment:
an error that relates to position close would limit the usage of this method only to position close scenarios. As far as I remember this method is intended to be used also in open lease and payment lease scenarios.